### DISCUSSION TITLE
TypeError: this is undefined

### DISCUSSION THREAD

**Discussion by ptman** [swc-project/swc#1575] [Q&A]
This is the code produced by swc that causes the problem:

```js
    onSend() {
      return _asyncToGenerator(function * () {
        if (this.msg === '') {
```

It used to work. I upgraded swc to 1.2.52 and it broke. I've tried downgrading to various versions.

**Comment by kdy1** [comment]
Can you provide the input code?

**Reply by ptman** [reply]
It was an async method in vue (2.x). I rewrote it to use promises. But it worked before. What changed?

So basically something like
```js
Vue.component('test', {
    methods: {
        async onSend() {
            if (this.msg === '') {
                ...
            }
        },
    },
});
```

**Comment by kdy1** [comment]
There was a pr related to this in async methods. I guess it's a regression caused by it.
